lib/ansible/modules/crypto/luks_device.py

```diff
@@ -0,0 +1,448 @@
+#!/usr/bin/python
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+    ---
+module: luks_device
+
+short_description: Manage encrypted (LUKS) devices
+
+version_added: "2.8"
+
+description:
+    - "Module manages LUKS on given device. Supports creating, destroying,
+      opening and closing of LUKS container and adding or removing new keys."
+
+options:
+    device:
+        description:
+            - "Device to work with (e.g. C(/dev/sda1)). Needed in most cases.
+              Can be omitted only when I(open=false) and I(name) is provided."
+        type: str
+    state:
+        description:
+            - "Based on its value creates or destroys the LUKS container on
+              a given device. Container creation requires I(device) and I(key)
+              options to be provided. Removal requires I(device) or I(name)."
+        default: present
+        choices: ['absent', 'present']
+        type: str
+    open:
+        description:
+            - "Based on its value unlocks or locks an existing LUKS container
+              on a given I(device). Use the I(name) option to set the name of
+              the opened container. Otherwise the name will be generated
+              automatically and returned in the result. Unlocking the container 
+              requires I(key) to be set."
+        default: false
+        type: bool
+    name:
+        description:
+            - "Sets container name when I(open=true). Can be used
+              instead of I(device) when closing the container
+              (i.e. when I(open=false))."
+        type: str
+    key:
```

Comment:
What do you think about changing this option name from `key` to `key_path`? If it is just `key`, I would expect to be able to put the key itself in here, and not a path to the keyfile.